src/3rdparty/squirrel/sqstdlib/sqstdaux.cpp

```diff
@@ -17,26 +17,21 @@ void sqstd_printcallstack(HSQUIRRELVM v)
 		SQBool b;
 		SQFloat f;
 		SQInteger level=1; //1 is to skip this function that is level 0
-		const SQChar *name=nullptr;
 		SQInteger seq=0;
 		pf(v,"\nCALLSTACK\n");
 		while(SQ_SUCCEEDED(sq_stackinfos(v,level,&si)))
 		{
-			const SQChar *fn="unknown";
-			const SQChar *src="unknown";
-			if(si.funcname)fn=si.funcname;
-			if(si.source) {
+			std::string_view fn="unknown";
+			std::string_view src="unknown";
+			if(!si.funcname.empty())fn=si.funcname;
+			if(!si.source.empty()) {
 				/* We don't want to bother users with absolute paths to all AI files.
 				 * Since the path only reaches NoAI code in a formatted string we have
 				 * to strip it here. Let's hope nobody installs openttd in a subdirectory
 				 * of a directory named /ai/. */
-				src = strstr(si.source, "\\ai\\");
-				if (!src) src = strstr(si.source, "/ai/");
-				if (src) {
-					src += 4;
-				} else {
-					src = si.source;
-				}
+				auto p = si.source.find("\\ai\\");
+				if (p == std::string_view::npos) p = si.source.find("/ai/");
+				src = p == std::string_view::npos ? si.source : si.source.substr(p + 4);
```

Comment:
```suggestion
				src = (p == std::string_view::npos) ? si.source : si.source.substr(p + 4);
```
Easier to read for me.